sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileScan.scala

```diff
@@ -65,6 +65,16 @@ trait FileScan extends Scan with Batch with SupportsReportStatistics with Loggin
    */
   def withPartitionFilters(partitionFilters: Seq[Expression]): FileScan
 
+  /**
+   * Returns the filters that can be use for file listing
+   */
+  def dataFilters: Seq[Expression]
```

Comment:
It seems too many methods to implement in `FileScan`.
How about just
```
def filters: Seq[Expression]
def withFilters(filters: Seq[Expression]): FileScan
```
And we get the data/partition filters in FileScan.

Reply:
ok, agree, I will make the necessary changes

Reply:
@gengliangwang it's seem to me more straight forward to split the filters to `dataFilters` and `partitionFilters` within the `PruneFileSourcePartitions` rule as there is a need to resolve the partition columns from the relation in order to get the `partitionFilters` and `dataFilters`.

Also from the point of view of the `FileScan` trait it seems cleaner to have as a metadata in the description distinction between the `dataFilters` and `partitionFilters`.

One option is to have 
```
def withFilters(partitionFilters: Seq[Expression], dataFilters: Seq[Expression]): FileScan
```
instead of 2 functions one for the `dataFilters` and one for the `partitionFilters`.

What do you think?

Reply:
> def withFilters(partitionFilters: Seq[Expression], dataFilters: Seq[Expression]): FileScan

this one seems good

Reply:
@gengliangwang fixed per the comment